### DISCUSSION TITLE
Loading SDXL models

### DISCUSSION THREAD

**Discussion by cibernicola** [vladmandic/sdnext#2458] [Q&A] [answered]
I'm not sure if I'm doing something wrong, very wrong, or if I've missed something, but I'm having a lot of trouble trying to load SDXL models and when I do, getting back to 1.5 models is an ordeal. I get errors everywhere, both when trying to use SDXL models and when going back to 1.5 models.

I have upgraded to  
`Version: app=sd.next updated=2023-11-03 hash=dc79cec1`
edit:
`updated: Version: app=sd.next updated=2023-11-06 hash=a2b50b69`

One error:
`Exception: Expected all tensors to be on the same device, but found at least two devices, cpu`

The settings are, I think, fine. But there is no way. The change between models should be relatively transparent, shouldn't it?
What am I doing wrong?
Thanks

Windows 10
RTX3090

**Comment by Tillerz** [comment, marked as answer]
Settings / User Interface, set this:

![image](https://github.com/vladmandic/automatic/assets/63370525/07e68c09-2a6e-49a2-954a-69bd0f4712e9)

Afterwards, you UI top looks lke this:

![image](https://github.com/vladmandic/automatic/assets/63370525/9271942f-a208-4728-ac96-f8226e4b68b5)

That's all you need to switch. Now, if you want to switch to SDXL, start at the right:

- set backend to Diffusers
- set pipeline to Stable Diffusion XL
- set a model/vae/refiner as needed
- RESTART THE UI
- see if everything stuck, if not, fix it
- Render images

There is some "autodetect" mode somewhere for some stuff, but I never tried that. The way above always works for switching between both backends.
